pkg/logql/log/filter.go

```diff
@@ -90,6 +89,68 @@ func (a andFilter) ToStage() Stage {
 	}
 }
 
+type andFilters struct {
+	filters []Filterer
+}
+
+// NewAndFilters creates a new filter which matches only if all filters match
+func NewAndFilters(filters []Filterer) Filterer {
+	containsAllFilter := containsAllFilter{}
+	regexpFilters := make([]Filterer, 0)
+	n := 0
+	for _, filter := range filters {
+		// Make sure we take care of panics in case a nil or noop filter is passed.
+		if !(filter == nil || filter == TrueFilter) {
+			switch c := filter.(type) {
+			case containsFilter:
+				// Join all contain filters.
+				containsAllFilter.Add(c)
+			case regexpFilter:
+				regexpFilters = append(regexpFilters, c)
+			default:
+				// Keep filter
+				filters[n] = filter
+				n++
+			}
+		}
+	}
+	filters = filters[:n]
+
+	if !containsAllFilter.Empty() {
+		filters = append(filters, containsAllFilter)
+	}
+
+	// Push regex filters to end
+	filters = append(filters, regexpFilters...)
+
+	if len(filters) == 0 {
+		return TrueFilter
```

Comment:
This should be at the top I think. Specially since you reslice line 117.

Reply:
I don't think so. We could have a list of `TrueFilter`s which will be thrown out by the reslicing. So we need to do this check afterwards.

Reply:
I see !